### PR TITLE
Add some component

### PR DESCRIPTION
**Pull request by hamed-sb**
Add
   +AutoComplete component
   +useAutoComplete hooks function
   +StylesProvider componen for support jss
   +some missing props

edit StyleBiulder for support static style

### CODE REVIEW

**Review by subroh0508 — CHANGES_REQUESTED** [review]
Sorry for the late review!

I tried to check the operation, but I got an error with the following code 😭  If it works in your local environment, I would like to see the sample code.

```kotlin
val data: Array<Data> = arrayOf(
    jsObject { name = "Mizuki Makabe"; age = 17 },
    jsObject { name = "Madoka Higuchi"; age = 17 },
)

autoComplete<Data> {
    attrs.options = data
    attrs.getOptionLabel = { it.name }
    attrs.renderInput = {
        buildElements {
            textField {
                console.log(Object.assign(attrs, it))
                Object.assign(attrs, it)
                attrs.label { +"Combo box" }
                attrs.variant = FormControlVariant.outlined
            }
        }
    }
}
```

Clicking on it will result in an error like this.
```
Uncaught TypeError: Cannot read property 'focus' of null
    at handleClick (webpack-internal:///../../node_modules/@material-ui/lab/esm/useAutocomplete/useAutocomplete.js:928)
    at HTMLUnknownElement.callCallback (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:237)
    at invokeGuardedCallback (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:292)
    at invokeGuardedCallbackAndCatchFirstError (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:306)
    at executeDispatch (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:389)
    at executeDispatchesInOrder (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:411)
    at executeDispatchesAndRelease (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:3278)
    at executeDispatchesAndReleaseTopLevel (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:3287)
    at forEachAccumulated (webpack-internal:///../../node_modules/react-dom/cjs/react-dom.development.js:3259)
```

**Comment by hamed-sb** [comment]
you must use props() method for set props
like this props( it as TextFieldProps)

sorry my English is not  good
thanks for review

File: sample/src/main/kotlin/demo/components/InputAdornmentsDemo.kt [hunk 1]
```diff
@@ -25,6 +31,15 @@ val ranges: List<Pair<String, String>>
...
     )
 
+external interface Person{
+    var name: String
+    var age: Int
+}
+val DATA: Array<Person> = arrayOf(
+    jsObject { name = "Mizuki Makabe"; age = 17 },
+    jsObject { name = "Madoka Higuchi"; age = 17 },
+)
```
Comment (subroh0508) [inline sample/src/main/kotlin/demo/components/InputAdornmentsDemo.kt:41]: 👍 👍 👍

**Review by subroh0508 — APPROVED** [review]
I think it's very good! I'm sorry I left it for two months 😭 
madoka

**Pull request merged**